--- a/components/bt/host/bluedroid/btc/profile/std/hf_ag/btc_hf_ag.c
+++ b/components/bt/host/bluedroid/btc/profile/std/hf_ag/btc_hf_ag.c
@@ -12,6 +12,7 @@
  * *
  ***********************************************************************************/
 #include <stdint.h>
+#include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
 #include <stdlib.h>
@@ -74,6 +75,7 @@
 #if (BTM_WBS_INCLUDED == TRUE)
 #ifndef BTC_HF_FEATURES
 #define BTC_HF_FEATURES    ( BTA_AG_FEAT_ECNR    | \
+                             BTA_AG_FEAT_3WAY    | \
                              BTA_AG_FEAT_REJECT  | \
                              BTA_AG_FEAT_ECS     | \
                              BTA_AG_FEAT_EXTERR  | \
@@ -87,6 +89,7 @@
 #ifndef BTC_HF_FEATURES
 #if BT_HF_AG_BQB_INCLUDED
 #define BTC_HF_FEATURES    ( BTA_AG_FEAT_REJECT  | \
+                             BTA_AG_FEAT_3WAY    | \
                              BTA_AG_FEAT_ECS     | \
                              BTA_AG_FEAT_EXTERR  | \
                              BTA_AG_FEAT_VREC    | \
@@ -95,6 +98,7 @@
                              BTA_AG_FEAT_UNAT )
 #else
 #define BTC_HF_FEATURES    ( BTA_AG_FEAT_ECNR    | \
+                             BTA_AG_FEAT_3WAY    | \
                              BTA_AG_FEAT_REJECT  | \
                              BTA_AG_FEAT_ECS     | \
                              BTA_AG_FEAT_EXTERR  | \
@@ -1483,6 +1487,20 @@
             do {
                 memcpy(param.unat_rep.remote_addr, &hf_local_param.btc_hf_cb[idx].connected_bda,sizeof(esp_bd_addr_t));
                 param.unat_rep.unat = p_data->val.str;
+                btc_hf_cb_to_app(ESP_HF_UNAT_RESPONSE_EVT, &param);
+            } while (0);
+            break;
+        }
+
+        case BTA_AG_AT_CHLD_EVT:
+        {
+            char chld_cmd[16];
+            idx = p_data->hdr.handle - 1;
+            CHECK_HF_IDX(idx);
+            do {
+                snprintf(chld_cmd, sizeof(chld_cmd), "+CHLD=%s", p_data->val.str);
+                memcpy(param.unat_rep.remote_addr, &hf_local_param.btc_hf_cb[idx].connected_bda,sizeof(esp_bd_addr_t));
+                param.unat_rep.unat = chld_cmd;
                 btc_hf_cb_to_app(ESP_HF_UNAT_RESPONSE_EVT, &param);
             } while (0);
             break;
