diff --git a/components/bt/host/bluedroid/btc/profile/std/a2dp/bta_av_co.c b/components/bt/host/bluedroid/btc/profile/std/a2dp/bta_av_co.c
--- a/components/bt/host/bluedroid/btc/profile/std/a2dp/bta_av_co.c
+++ b/components/bt/host/bluedroid/btc/profile/std/a2dp/bta_av_co.c
@@ -82,7 +82,7 @@ const tA2D_SBC_CIE bta_av_co_sbc_caps = {
 
 /* SBC SINK codec capabilities */
 const tA2D_SBC_CIE bta_av_co_sbc_sink_caps = {
-    (A2D_SBC_IE_SAMP_FREQ_48 | A2D_SBC_IE_SAMP_FREQ_44), /* samp_freq */
+    (A2D_SBC_IE_SAMP_FREQ_44), /* samp_freq */
     (A2D_SBC_IE_CH_MD_MONO | A2D_SBC_IE_CH_MD_STEREO | A2D_SBC_IE_CH_MD_JOINT | A2D_SBC_IE_CH_MD_DUAL), /* ch_mode */
     (A2D_SBC_IE_BLOCKS_16 | A2D_SBC_IE_BLOCKS_12 | A2D_SBC_IE_BLOCKS_8 | A2D_SBC_IE_BLOCKS_4), /* block_len */
     (A2D_SBC_IE_SUBBAND_4 | A2D_SBC_IE_SUBBAND_8), /* num_subbands */
@@ -403,10 +403,10 @@ void bta_av_build_src_cfg (UINT8 *p_pref_cfg, UINT8 *p_src_cap)
         return ;
     }
 
-    if (src_cap.samp_freq & A2D_SBC_IE_SAMP_FREQ_48) {
-        pref_cap.samp_freq = A2D_SBC_IE_SAMP_FREQ_48;
-    } else if (src_cap.samp_freq & A2D_SBC_IE_SAMP_FREQ_44) {
+    if (src_cap.samp_freq & A2D_SBC_IE_SAMP_FREQ_44) {
         pref_cap.samp_freq = A2D_SBC_IE_SAMP_FREQ_44;
+    } else if (src_cap.samp_freq & A2D_SBC_IE_SAMP_FREQ_48) {
+        pref_cap.samp_freq = A2D_SBC_IE_SAMP_FREQ_48;
     }
 
     if (src_cap.ch_mode & A2D_SBC_IE_CH_MD_JOINT) {
